### PR TITLE
D3D: Fix Dolphin immediately exiting exclusive fullscreen.

### PR DESCRIPTION
**Pull request by Armada651**
This PR fixes an issue reported on the forums.

I also added an OSD message to make it clear to the user when exactly exclusive mode is enabled.

### CODE REVIEW

**Comment by Armada651** [comment]
@degasus Please review.

File: Source/Core/VideoBackends/D3D/Render.cpp [hunk 1]
```diff
@@ -885,26 +885,21 @@ void Renderer::SwapImpl(u32 xfbAddr, u32 fbWidth, u32 fbStride, u32 fbHeight, co
 	{
 		if (fullscreen && !exclusive_mode)
 		{
+			g_Config.bExclusiveMode = false;
+
+			if (g_Config.bExclusiveMode)
```
Comment (degasus) [inline Source/Core/VideoBackends/D3D/Render.cpp]: This is always false

Comment (Armada651) [inline Source/Core/VideoBackends/D3D/Render.cpp]: Oops, should've been in a different other, thanks for spotting it.

File: Source/Core/DolphinWX/Frame.cpp [hunk 2]
```diff
@@ -211,6 +211,11 @@ bool CRenderFrame::ShowFullScreen(bool show, long style)
 	{
 		// OpenGL requires the pop-up style to activate exclusive mode.
 		SetWindowStyle((GetWindowStyle() & ~wxDEFAULT_FRAME_STYLE) | wxPOPUP_WINDOW);
+
+		// Some backends don't support exclusive fullscreen, so we
+		// can't tell exactly when exclusive mode is activated.
+		if (!g_Config.backend_info.bSupportsExclusiveFullscreen)
+			OSD::AddMessage("Enabled exclusive fullscreen.");
```
Comment (degasus) [inline Source/Core/DolphinWX/Frame.cpp]: Is it really enabled when it's unsupported?

**Comment by degasus** [comment]
Isn't it a bit noisy to generate an OSD for every fullscreen enabling/disabling?

File: Source/Core/DolphinWX/Frame.cpp (hunk 2 above)
Comment (Armada651) [inline Source/Core/DolphinWX/Frame.cpp]: The drivers should enable exclusive mode automatically when the relevant flags are set. But we can't tell exactly when it is active.

**Comment by Armada651** [comment]
@degasus It is noisy, that's why I was reluctant to add such messages. But there are no clear signs to the user exactly when exclusive mode is enabled. The only real clues are only detectable when you have two monitors.

I could also just not tell them, which is what I actually prefer anyway.

**Pull request merged**